apps/frontend/src/app/3_organisms/TransactionStepDialog/components/TransactionStep/TransactionStep.tsx

```diff
@@ -0,0 +1,262 @@
+import React, { FC, useCallback, useMemo, useState } from 'react';
+
+import { formatUnits, parseUnits } from 'ethers/lib/utils';
+import { bignumber } from 'mathjs';
+
+import {
+  Accordion,
+  AmountInput,
+  Button,
+  ButtonType,
+  ButtonStyle,
+  Heading,
+  HeadingType,
+  Paragraph,
+  RadioButtonGroup,
+  SimpleTable,
+  SimpleTableRow,
+  StatusItem,
+  StatusType,
+  TransactionId,
+  HelperButton,
+} from '@sovryn/ui';
+
+import { chains, defaultChainId } from '../../../../../config/chains';
+import { tokens } from '../../../../../config/tokens';
+import { Transaction, TxConfig } from '../../TransactionStepDialog.types';
+
+export type TransactionStepProps = {
+  transaction: Transaction;
+  step: string | number;
+  status: StatusType;
+  config: TxConfig;
+  updateConfig: (config: TxConfig) => void;
+  gasPrice: string;
+  isLoading: boolean;
+};
+
+const chain = chains.find(chain => chain.id === defaultChainId);
+
+export const TransactionStep: FC<TransactionStepProps> = ({
+  step,
+  status,
+  transaction,
+  config,
+  gasPrice,
+  updateConfig,
+  isLoading,
+}) => {
+  const token = useMemo(
+    () =>
+      tokens.find(
+        token =>
+          token.address.toLowerCase() ===
+          transaction.contract.address.toLowerCase(),
+      ),
+    [transaction.contract.address],
+  );
+
+  const { title, subtitle } = transaction;
+
+  const resetConfig = useCallback(async () => {
+    try {
+      const gasLimit = await transaction.contract.estimateGas[
+        transaction.fnName
+      ](...transaction.args);
+
+      updateConfig({
+        ...transaction.config,
+        unlimitedAmount: config.unlimitedAmount,
+        amount:
+          transaction.fnName === 'approve' ? transaction.args[1] : undefined,
```

Comment:
Extract "approve" string to `apps/frontend/src/utils/constants.ts` for reuse - we may want enum for the various expected tx types